rotkehlchen/api/v1/schemas.py

```diff
@@ -1670,32 +1670,51 @@ class ExternalServicesResourceDeleteSchema(Schema):
     services = fields.List(SerializableEnumField(enum_class=ExternalService), required=True)
 
 
-class ExchangesResourceEditSchema(Schema):
-    name = NonEmptyStringField(required=True)
+class _BinanceMarketsSchema(Schema):
+    """Common logic used when adding and editing exchange credentials"""
     location = LocationField(limit_to=SUPPORTED_EXCHANGES, required=True)
+    binance_markets = fields.List(NonEmptyStringField, load_default=None)
+
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        if (
+            data['location'] in (Location.BINANCE, Location.BINANCEUS) and
+            (data['binance_markets'] is None or len(data['binance_markets']) == 0)
+        ):
+            raise ValidationError(
+                message='Binance API key requires at least one market pair to be selected. '
+                'Please choose the trading pairs you want to monitor before adding the API key.',
+                field_name='binance_markets',
+            )
+
+
+class ExchangesResourceEditSchema(_BinanceMarketsSchema):
```

Comment:
I don't understand why you made another class. Can you explain this to me? It's not as if we have different classes depending on the type of exchange. There is no point in this separation or am I missing something?

Reply:
as explained in discord is to avoid repeating the same validation both when adding an exchange or editing it 
<img width="678" height="755" alt="image" src="https://github.com/user-attachments/assets/ca8b4595-1a14-47e1-aed5-88c0b04d68da" />